inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -431,10 +431,12 @@ void MKLDNNPlugin::MKLDNNInferRequest::SetBlob(const std::string& name, const In
                     IE_THROW(ParameterMismatch) << "Failed to set input blob. Blocking descriptor mismatch.";
             }
 
-            const auto &actualDesc = graph->getInputNodeByName(name)->getChildEdgesAtPort(0)[0]->getMemory().getDesc();
+            const auto actualDesc = MemoryDescUtils::convertToBlockedMemoryDesc(graph->getInputNodeByName(name)->getBaseMemDescAtOutputPort(0));
```

Comment:
What if we make defined descriptor from this udefined one with shape from the `blobDesc`, than we can just call `isCompatible` for the descriptors. What do you think about this approach?

Reply:
`cloneWithNewDims` contain inside
1. checks shapes on compatibility
2. check dims on UNDEF value
3. check strides on dense
4. strides and blocked dims recomputation
5. and new descriptor creation

looks like we got too much overhead

Reply:
But with such approach the code will be way compact and understandable. Given that the size of the arrays are about 5-6 elements, I would say this overhead is negligible. At least we can measure it and compare with the existing algorithm. Because all the states above have O(n) complexity and I would say comparable in time with conversion to blocked mem desc for example.

Reply:
fixed